backend/src/blueprints/security/auth0_service.py

```diff
@@ -0,0 +1,62 @@
+from http import HTTPStatus
+
+import jwt
+
+from src.utils import json_abort
+
+
+class Auth0Service:
+    """Perform JSON Web Token (JWT) validation using PyJWT"""
+
+    def __init__(self):
+        self.issuer_url = None
+        self.audience = None
+        self.algorithm = "RS256"
+        self.jwks_uri = None
+
+    def initialize(self, auth0_domain, auth0_audience):
+        self.issuer_url = f"https://{auth0_domain}/"
+        self.jwks_uri = f"{self.issuer_url}.well-known/jwks.json"
+        self.audience = auth0_audience
+
+    def get_signing_key(self, token):
+        try:
+            jwks_client = jwt.PyJWKClient(self.jwks_uri)
+
+            return jwks_client.get_signing_key_from_jwt(token).key
+        except Exception as error:
+            json_abort(
```

Comment:
is there a reason why we're not using the message class we defined here? i figure we'd want to keep things consistent.